packages/iov-tendermint-rpc/src/rpcclients/websocketclient.ts

```diff
@@ -212,4 +213,13 @@ export class WebsocketClient implements RpcStreamingClient {
   protected async responseForRequestId(id: JsonRpcId): Promise<JsonRpcResponse> {
     return firstEvent(this.jsonRpcResponseStream.filter(r => r.id === id));
   }
+
+  private reconnectedHandler(): void {
+    [...this.subscriptionStreams.keys()].forEach(query => {
+      const request = createJsonRpcRequest("subscribe", { query: query });
+      const producer = new RpcEventProducer(request, this.socket);
+      const stream = Stream.create(producer);
+      this.subscriptionStreams.set(query, stream);
```

Comment:
This sets `subscriptionStreams[subscriptionStreams]` to a new reference. The old stream the caller got via `const eventStream = this.client.listen(req);` will not change, so the new events do not arrive.

Also this re-establishes subscribtions that nobody listens to anymore (where the original producer was stopped).

My feeling is that in `public listen(request: JsonRpcRequest): Stream<SubscriptionEvent> {` there needs to be a producer (maybe a wrapper around `RpcEventProducer`) which survives the reconnect and routes events from different sources.